src/Service/Modules/Reports/PurchaseOrdersReportModule.cs

```diff
@@ -11,23 +14,35 @@
     using Carter.Response;
 
     using Linn.Common.Facade;
+    using Linn.Common.Serialization;
     using Linn.Purchasing.Facade.Services;
     using Linn.Purchasing.Resources;
     using Linn.Purchasing.Service.Extensions;
     using Linn.Purchasing.Service.Models;
 
     using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Http.Extensions;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.AspNetCore.Razor.Language.CodeGeneration;
+    using Microsoft.Extensions.FileProviders;
     using Microsoft.Extensions.Primitives;
 
     public class PurchaseOrdersReportModule : CarterModule
     {
         private readonly IPurchaseOrderReportFacadeService purchaseOrderReportFacadeService;
 
+        private readonly CsvStreamWriter csvStreamWriter;
+
+        private MemoryStream stream;
```

Comment:
I would declare a

Reply:
stream in a using block

Reply:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement

Reply:
something like this? I'm not sure I can use dependency injection for the csvStreamWriter because I need to give it the stream when I new it up

`  using (MemoryStream stream = new MemoryStream())
            {
                CsvStreamWriter csvStreamWriter = new CsvStreamWriter(stream);
                csvStreamWriter.WriteModel(results);
                var contentDisposition = new ContentDisposition
                                             {
                                                 FileName =
                                                     $"ordersBySupplier{resource.From.Substring(0, 10)}_To_{resource.To.Substring(0, 10)}.csv"
                                             };

                this.stream.Position = 0;
                await res.FromStream(this.stream, "text/csv", contentDisposition);
            }`

Reply:
That'll work. I think as of C# 8 you can omit the curly brackets and just delcare it like a normal var, e.g.

using var stream = new MemoryStream(); 

Werdly my visual studio complains when I do that, but its definitely legal syntax and it compiles

Reply:
except this.stream.Position = 0 will need to just be stream.Position... now obviously